pilot/pkg/bootstrap/istio_ca.go

```diff
@@ -568,3 +589,42 @@ func (s *Server) createIstioRA(opts *caOptions) (ra.RegistrationAuthority, error
 	})
 	return raServer, err
 }
+
+// checkCABundleCompleteness checks if all required CA certificate files exist
+func checkCABundleCompleteness(signingKeyFile, signingCertFile, rootCertFile string, chainFiles []string) (signingCABundleComplete bool, bundleExists bool) {
+	signingKeyExists := fileExists(signingKeyFile)
+	signingCertExists := fileExists(signingCertFile)
+	rootCertExists := fileExists(rootCertFile)
+	chainFilesExist := hasValidChainFiles(chainFiles)
+
+	bundleExists = signingKeyExists || signingCertExists || rootCertExists || chainFilesExist
+	signingCABundleComplete = signingKeyExists && signingCertExists && rootCertExists && chainFilesExist
+
+	return signingCABundleComplete, bundleExists
+}
+
+// fileExists checks if a file exists and is accessible
+func fileExists(filename string) bool {
+	if filename == "" {
+		return false
+	}
+	_, err := os.Stat(filename)
+	if err != nil && !os.IsNotExist(err) {
+		log.Warnf("Unexpected error checking file %s: %v", filename, err)
+	}
+	return err == nil
```

Comment:
nit - for the sake of keeping the function more flexible in the future, why not return the error and then let the caller determine what to do with it? Like the caller could be responsible for any retry logic, failure, or other error processing if it knows that there's an error vs if the file doesn't actually exist.

This would entail having two return values though, both the boolean and the error value. Not super important just a thought.